pypesto/ensemble/ensemble.py

```diff
@@ -256,6 +321,37 @@ def _compute_summary(tmp_array, percentiles_list):
         # also return the object
         return self.prediction_summary
 
+    def compute_chi2(self, amici_objective: AmiciObjective):
+        if (self.prediction_summary[MEAN] is None) or (
+                self.prediction_summary[WEIGHTED_SIGMA] is None):
+            try:
+                self.compute_summary(
+                        weighting=True,
+                        compute_weighted_sigma=True)
+            except TypeError:
+                raise ValueError('Computing a summary failed.')
+        n_conditions = len(self.prediction_results[0].conditions)
+        chi_2 = []
+        for i_cond in range(n_conditions):
+            # get measurements and put into right form
+            y_meas = amici_objective.edatas[i_cond].getObservedData()
+            y_meas = np.array(y_meas)
+            # bring into shape (n_t,n_y)
+            y_meas = y_meas.reshape(
+                (int(len(y_meas)/amici_objective.edatas[0].nytrue()),
```

Comment:
Can add a comment e.g. `# assumes all observables are provided at all time points`.

Could you replace this with e.g. `amici_objective.edatas[0].nt()`?
https://github.com/AMICI-dev/AMICI/blob/509c850b57dd4197253d0cf30ffb4ba04c752d05/include/amici/edata.h#L145